### ISSUE TITLE
Feature request: Add way to full-reload all shaders, ignoring time stamp.

### ISSUE THREAD

**Issue by bo3b** [bo3b/3Dmigoto#26]
For use of #include files for common code, changing the #include file does not reload all shaders that use that file.  

Request to have a path to ignore time stamps and reload all files, maybe by alternate key, or modifier like Shift to F10. 

Might make more sense to make it a .ini preference for hunting, rather than something that changes on a per file basis.

**Comment by oomek** [comment]
To avoid unnecessary directory crawling a separate ini section containing hashes of shaders which 3DM should always reload regardless of the timestamp would be a compromise maybe.

**Comment by DarkStarSword** [comment]
We should add an #include handler to 3DMigoto rather than using the default. That will allow us to more consistently handle includes for games that change their current working directory (e.g. Witcher 3) and track what shaders need to be reloaded when an included file is updated.

We should refactor the code first though, I think we currently have four places where we call D3DCompile.

**Comment by bo3b** [comment]
Yes, sorry, but this is lowish priority compared to some of the other serious problems with the code.  It works, but no one will say it's a gem.

Lots of offline workaround possibilities for this problem.  Scripts to 'touch' files on demand, cmd scripts that can monitor any touch of include file.  Macros in NotePad++ to touch every file open.

**Comment by oomek** [comment]
It's been a year since that request was opened. Did it advance in queue of pending features by any chance?

**Comment by DarkStarSword** [comment]
You just need to get me to work on enough games where this is a serious annoyance and eventually I'll do something about it ;-)
